lib/p2p/Peer.ts

```diff
@@ -604,69 +644,126 @@ class Peer extends EventEmitter {
   }
 
   /**
-   * Sends a hello packet and waits for one to be received, if we haven't received a hello packet already.
+   * Authenticate the identity of the peer through SessionInit packet
+   * @param {SessionInitPacket} packet
+   * @param {NodeKey} nodeKey
    */
-  private initHello = async (handshakeData: HandshakeState) => {
-    const packet = new packets.HelloPacket(handshakeData);
+  private authenticate = (packet: packets.SessionInitPacket, nodeKey: NodeKey) => {
+    const body = packet.body!;
+    const { sign, ...bodyWithoutSign } = body;
+    const { nodePubKey } = body.nodeState; // the peer pubkey
+    const { peerPubKey } = body; // our own pubkey
 
-    this.sendPacket(packet);
+    // verify that the msg was signed the for us
```

Comment:
Can drop "the" in "was signed the for us", and maybe say "was intended for us" since it's possible it wasn't signed at all